apps/flame-defi/app/swap/hooks/use-swap-button.tsx

```diff
@@ -346,6 +353,7 @@ export function useSwapButton({
     txnStatus,
     setTxnStatus,
     txnMsg,
-    tokenApprovalNeeded: tokenApprovalNeeded !== null,
+    tokenApprovalNeeded:
```

Comment:
I don't think we should add extra logic to this expression. It makes the exported property name incorrect. Should we just add the checks for pending at the parent? We already export `txnStatus`.

I also think it's a bit confusing to use `tokenApprovalNeeded` as a boolean in one area, but as an `EvmCurrency` in other areas.

Can we pick some different variable names that would make all this more clear?

Reply:
`validSwapInputs` also already includes `txnStatus !== TXN_STATUS.PENDING` in its expression.

That's all very intertwined, especially since in the parent there are expressions both like
`!validSwapInputs && !tokenApprovalNeeded` and `validSwapInputs && !tokenApprovalNeeded`.

I think there's room to simplify

Reply:
What if we rename tokenApprovalNeeded to tokenNeedingApproval in the use-swap-button hook so it is clear that it is a EvmCurrency or null. 

And we keep the tokenApprovalNeeded export as a boolean and keep the check as it is here. Checking if tokenApprovalNeeded is not null and the txnStatus is not pending. Having to sprinkle around the txnStatus pending check in the parent is more confusing imo.